File: frontend/src/model/riddle.ts
export interface Riddle {
  id: number;
  title: string;
  description: string;
  code: string;
  testCases: TestCase[];
  validationCode: string;
  functionName?: string;
  author?: Author;
}

export interface TestCase {
  input: string;
  output: string;
}

export interface Author {
  username: string;
}

File: backend/src/main/java/com/zurakin/codingriddles/models/Riddle.java
package com.zurakin.codingriddles.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;


@Entity
@Table(name="riddles")
@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Riddle {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Lob
    private String description;
    private String code;
    private String validationCode;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "riddle_id")
    private List<TestCase> testCases;
}
